experimenter/experimenter/experiments/tests/api/v8/test_serializers.py

```diff
@@ -0,0 +1,372 @@
+from experimenter.experiments.models import NimbusBranchFeatureValue, NimbusExperiment
+from experimenter.experiments.tests.factories import (
+    TEST_LOCALIZATIONS,
+    NimbusBranchFactory,
+    NimbusExperimentFactory,
+    NimbusFeatureConfigFactory,
+)
+
+
+class TestNimbusExperimentSerializer(TestCase):
+    maxDiff = None
+
+    def test_expected_schema_with_desktop(self):
+        locale_en_us = LocaleFactory.create(code="en-US")
+        application = NimbusExperiment.Application.DESKTOP
+        feature1 = NimbusFeatureConfigFactory.create(application=application)
+        feature2 = NimbusFeatureConfigFactory.create(application=application)
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
+            application=application,
+            firefox_min_version=NimbusExperiment.MIN_REQUIRED_VERSION,
+            feature_configs=[feature1, feature2],
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
+            channel=NimbusExperiment.Channel.NIGHTLY,
+            primary_outcomes=["foo", "bar", "baz"],
+            secondary_outcomes=["quux", "xyzzy"],
+            locales=[locale_en_us],
+            _enrollment_end_date=datetime.date(2022, 1, 5),
+        )
+        serializer = NimbusExperimentSerializer(experiment)
+        experiment_data = serializer.data.copy()
+        bucket_data = dict(experiment_data.pop("bucketConfig"))
+        branches_data = [dict(b) for b in experiment_data.pop("branches")]
+        feature_ids_data = experiment_data.pop("featureIds")
+
+        assert experiment.start_date
+        assert experiment.actual_enrollment_end_date
+        assert experiment.end_date
+
+        min_required_version = NimbusExperiment.MIN_REQUIRED_VERSION
+
+        self.assertDictEqual(
+            experiment_data,
+            {
+                "arguments": {},
+                "application": "firefox-desktop",
+                "appName": "firefox_desktop",
+                "appId": "firefox-desktop",
+                "channel": "nightly",
+                # DRF manually replaces the isoformat suffix so we have to do the same
+                "startDate": experiment.start_date.isoformat().replace("+00:00", "Z"),
+                "enrollmentEndDate": (
+                    experiment.actual_enrollment_end_date.isoformat().replace(
+                        "+00:00", "Z"
+                    )
+                ),
+                "endDate": experiment.end_date.isoformat().replace("+00:00", "Z"),
+                "id": experiment.slug,
+                "isEnrollmentPaused": True,
+                "isRollout": False,
+                "proposedDuration": experiment.proposed_duration,
+                "proposedEnrollment": experiment.proposed_enrollment,
+                "referenceBranch": experiment.reference_branch.slug,
+                "schemaVersion": settings.NIMBUS_SCHEMA_VERSION,
```

Comment:
We should probably do something different with this setting as well. Maybe create a NIMBUS_ANALYSIS_SCHEMA_VERSION and just set it to 1.0.0 for now. @jaredlockhart any suggestions here?

Reply:
Oh yeah that's there solely for client behaviour, probably has no meaning here.  Having a locked schema like that is only useful for supporting legacy clients to prevent breakage.  I don't expect we'd need anything similar for jetstream since we control both sides and there's no case where we'd have 'legacy jetstreams' running we're not able to update.

Reply:
Ok, I'd say we just leave the existing setting alone then for now. We can rip it out entirely later, no reason to create another useless setting in the meantime.